workers/unity/Packages/io.improbable.gdk.tools/GenerateCode.cs

```diff
@@ -476,5 +450,91 @@ private static IEnumerable<string> FindDirInPackages(string directory)
+                $"--log-file=\"{logfilePath}\""
+            };
+
+            if (toolsConfig.VerboseLogging)
+            {
+                codegenArgs.Add($"--verbose");
+            }
+
+            var codegenOutputPath = Path.GetFullPath(Path.Combine(Application.dataPath, "..", toolsConfig.CodegenOutputDir));
+            codegenArgs.Add($"--native-output-dir=\"{codegenOutputPath}\"");
+
+            // Add user defined schema directories
+            codegenArgs.AddRange(toolsConfig.SchemaSourceDirs
+                .Where(Directory.Exists)
+                .Select(directory => $"--schema-path=\"{Path.GetFullPath(directory)}\""));
+
+            // Add package schema directories
+            codegenArgs.AddRange(FindDirInPackages(SchemaPackageDir)
+                .Select(directory => $"--schema-path=\"{directory}\""));
+
+            // Schema Descriptor
+            codegenArgs.Add($"--descriptor-dir=\"{toolsConfig.DescriptorOutputDir}\"");
+
+            codegenArgs.AddRange(
+                toolsConfig.SerializationOverrides.Select(@override => $"--serialization-override=\"{@override}\""));
+
+            var codegenArgsString = string.Join(" ", codegenArgs);
+
+            // For dotnet run / visual studio
+            try
+            {
+                var csprojXml = XDocument.Load(CodegenExe);
+                var projectNode = csprojXml.Element("Project");
+                var propertyGroup = projectNode.Element("PropertyGroup");
+
+                var args = propertyGroup.Element("StartArguments");
+                args?.Remove();
+
+                propertyGroup.Add(XElement.Parse($"<StartArguments>{codegenArgsString}</StartArguments>"));
+                csprojXml.Save(CodegenExe);
+            }
+            catch (Exception e)
+            {
+                Debug.LogError("Unable to update csproj with run configuration.");
+                Debug.LogException(e);
+                throw;
+            }
+
+            // For jetbrains rider
+            try
+            {
+                var runConfigPath = Path.Combine(CodegenExeDirectory, ".idea", ".idea.CodeGen", ".idea",
+                    "runConfigurations");
+                Directory.CreateDirectory(runConfigPath);
+                using (var w = new StreamWriter(Path.Combine(runConfigPath, "CodeGen.xml"), false))
+                {
+                    w.Write($@"
+<component name=""ProjectRunConfigurationManager"">
+  <configuration default=""false"" name=""CodeGen"" type=""DotNetProject"" factoryName="".NET Project"">
+    <option name=""PROJECT_PATH"" value=""$PROJECT_DIR$/CodeGen/CodeGen.csproj"" />
+    <option name=""PROGRAM_PARAMETERS"" value=""{codegenArgsString.Replace("\"", "&quot;")}"" />
+  </configuration>
+</component>
+");
+                }
+            }
+            catch (Exception e)
+            {
+                Debug.LogError("Unable to generate Rider run configuration.");
+                Debug.LogException(e);
+                throw;
```

Comment:
Will this exception effectively be logged twice because of the re-throw?

Reply:
yes - though I'll now change it to just throw an Exception with this context and catch/log it at a higher level